cmd/bugzilla-backporter/main.go

```diff
@@ -0,0 +1,187 @@
+		labels := prometheus.Labels{"status": strconv.Itoa(trw.statusCode), "path": r.URL.EscapedPath()}
+		bzbpMetrics.httpRequestDuration.With(labels).Observe(latency.Seconds())
+		bzbpMetrics.httpResponseSize.With(labels).Observe(float64(trw.size))
+	})
+}
+
+// Needs additional work
+func getAuthToken() []byte {
+	// Auth token commented out for security reasons
+	// Will modify function to use serviceAgent
+	return []byte("")
+}
+
+// Handler returns a function which populates the response with the details of the bug
+// Expects a POST request with field ID
+// Returns bug details in JSON format
+func getBugHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			w.WriteHeader(http.StatusNotImplemented)
+			w.Write([]byte(http.StatusText(http.StatusNotImplemented)))
+			return
+		}
+
+		encoded, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Could not read bug ID from request body."))
+			return
+		}
+		client := bugzilla.NewClient(getAuthToken, "https://bugzilla.redhat.com/")
+		// Reusing the bug struct from bugzilla namespace
+		bug := bugzilla.Bug{}
+		if err = json.Unmarshal(encoded, &bug); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Could not parse request body as unresolved config."))
+			return
+		}
+		bugInfo, err := client.GetBug(bug.ID)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte("Bug ID not found"))
+			return
+		}
+		jsonBugInfo, err := json.MarshalIndent(*bugInfo, "", "  ")
+		if err != nil {
+			recordError("failed to marshal bugInfo")
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "failed to marshal bugInfo to JSON: %v", err)
+			// logger.WithError(err).Errorf("failed to marshal bugInfo to JSON")
+			return
+		}
+		fmt.Println(bugInfo)
+		w.WriteHeader(http.StatusOK)
+		w.Write(jsonBugInfo)
+	}
+}
+
+func genericHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusNotFound)
+		w.Write([]byte(http.StatusText(http.StatusNotFound)))
+	}
+}
+
+func gatherOptions() options {
+	o := options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.logLevel, "log-level", "info", "Level at which to log output.")
+	fs.StringVar(&o.address, "address", ":8080", "Address to run server on")
+	fs.StringVar(&o.uiAddress, "ui-address", ":8082", "Address to run the registry UI on")
+	fs.DurationVar(&o.gracePeriod, "gracePeriod", time.Second*10, "Grace period for server shutdown")
+	fs.Parse(os.Args[1:])
+	return o
+}
+
+func validateOptions(o options) error {
+	_, err := log.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %v", err)
+	}
+	return nil
+}
+
+func main() {
+	o := gatherOptions()
+	err := validateOptions(o)
+	if err != nil {
+		log.Fatalf("invalid options: %v", err)
+	}
+	level, _ := log.ParseLevel(o.logLevel)
```

Comment:
That is being checked in the validateOptions function, but i'm not exiting at that point. Do you thing it would be a better idea to fail and exit at that point?

Reply:
I think I would just avoid doing it twice, maybe rename `validateOptions` to `completeOptions` and set the log level right there (unless the parsing errored, ofc)

Reply:
i think the intention with the validate options was to leave room to extend it for the rest of the options... but I guess its overkill since I'm not expecting that many options, at least at this point.